### PR TITLE
Site Settings: Move Posts Per Page field to Custom Content Types

### LINKED ISSUE #14910: CPTs: Move Posts Per Page field to Custom Content Types
As suggested in #13568, let's move the regular "Posts per Page" field to the Custom Content Types card and rename the card to "Content Types.

### PR DESCRIPTION
**Pull request by tyxla**
### Purpose

As suggested in #13568, this PR moves the regular **Posts per Page** field to the **Custom Content Types** card and renames that card to **Content Types**. 

### Rationale

Rationale behind this change is simple: all posts per page fields to be managed in the same place, and in a consistent way. A good consequence from that change is that we're able to remove some code, because we're reusing what we can from the current **Custom Content Types** card.

Fixes #14910.

### Preview

**After (the new Custom Content Types card)**

**Before (the old Posts Per Page field in the Composing card)**

**Before (the old Custom Content Types card)**

### To test
* Checkout this branch, or get it going on calypso.live.
* Go to `/settings/writing/:site`, where `:site` is a Jetpack site.
* Verify you see the Posts Per Page field in the new location (**Custom Content Types** card), and not in the old one (**Composing** card).
* Verify the new **Posts Per Page** field saves and retrieves the value correctly.
* Test the above for a WP.com site.

### CODE REVIEW

File: client/my-sites/site-settings/custom-content-types/index.jsx [hunk 1]
```diff
@@ -124,6 +134,18 @@ class CustomContentTypes extends Component {
 		);
 	}
 
+	renderBlogPostSettings() {
+		const { translate } = this.props;
+		const fieldLabel = translate( 'Blog posts' );
```
Comment (a8ci18n) [inline client/my-sites/site-settings/custom-content-types/index.jsx]: Hi! I've found a possible matching string that has already been translated **57** times:
`translate( 'Blog Posts' )` *ES Score: 10*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/site-settings/form-writing.jsx [hunk 2]
```diff
@@ -126,7 +126,7 @@ class SiteSettingsFormWriting extends Component {
 				}
 
 				{
-					this.renderSectionHeader( translate( 'Custom content types' ) )
+					this.renderSectionHeader( translate( 'Content types' ) )
```
Comment (a8ci18n) [inline client/my-sites/site-settings/form-writing.jsx]: Hi! I've found a possible matching string that has already been translated **28** times:
`translate( 'Content Types' )` *ES Score: 13*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/move-posts-per-page-to-cpt)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

Comment (tyxla) [inline client/my-sites/site-settings/form-writing.jsx]: Nice tip @a8ci18n, but we're going with the lowercase one intentionally - it was a design decision a while ago. Should we change the original translations, @yoavf?

File: client/my-sites/site-settings/custom-content-types/index.jsx (hunk 1 above)
Comment (tyxla) [inline client/my-sites/site-settings/custom-content-types/index.jsx]: Nice tip @a8ci18n, but we're going with the lowercase one intentionally - it was a design decision a while ago. Should we change the original translations, @yoavf?

**Comment by seear** [comment]
Should we change the name of `renderToggle()` since it always did more than that and now sometimes doesn't even render a toggle? Or split things differently so that it _only_ renders a toggle?

File: client/my-sites/site-settings/form-writing.jsx (hunk 2 above)
Comment (yoavf) [inline client/my-sites/site-settings/form-writing.jsx]: The existing translations are for an obsolete string (a bit of an edge case here), so nothing to change elsewhere. We should be able to reuse the existing translations when this is merged.

**Comment by seear** [comment]
Also I would probably wait on some design input on this, the indenting combined with the same font weight for 'Blog posts', and 'Display...' don't look quite right to me:

screen shot 2017-07-17 at 12 38 19

**Comment by tyxla** [comment]
> My only concern would be the `name !== 'post'` line in `renderToggle()`, and I think changing the name of the functions would make that ok, but I'll leave it up to you.

Good call! We no longer need the `renderToggle` as a separate method anyway, so I've merged them together with `renderContentTypeSettings()`.

> Also I would probably wait on some design input on this, the indenting combined with the same font weight for 'Blog posts', and 'Display...' don't look quite right to me:

Well, I only followed what was suggested here: https://github.com/Automattic/wp-calypso/issues/13568#issuecomment-300597423 but I'm open to feedback - as you say, let's wait for design folks to chime in.

In the meantime, @seear, the PR is ready for another code review after I addressed your recommendation.

File: client/my-sites/site-settings/custom-content-types/index.jsx [hunk 3]
```diff
@@ -116,10 +95,45 @@ class CustomContentTypes extends Component {
...
+				<FormSettingExplanation isIndented>
+					{ description }
+				</FormSettingExplanation>
+			</div>
+		);
+	}
+
+	renderBlogPostSettings() {
+		const { translate } = this.props;
+		const fieldLabel = translate( 'Blog posts' );
```
Comment (a8ci18n) [inline client/my-sites/site-settings/custom-content-types/index.jsx]: Hi! I've found a possible matching string that has already been translated **57** times:
`translate( 'Blog Posts' )` *ES Score: 10*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by seear** [comment]
> Well, I only followed what was suggested here: #13568 (comment)

Ah yeah. Seems there is a sliver more spacing there underneath _Blog posts_?

**Comment by tyxla** [comment]
> Seems there is a sliver more spacing there underneath Blog posts?

Good 👁,  @seear. Fixed in https://github.com/Automattic/wp-calypso/pull/16281/commits/04d7f3333a004dff5b74bd57270572767dc8b96f, here is an updated preview:

**Comment by tyxla** [comment]
@rickybanister does this look good to go?

**Comment by rickybanister** [comment]
I think it could benefit from bolding the labels. I know that our toggle labels are not normally bolded, but because 'Blog posts' does not have a toggle the whole thing looks a little bit off. 

Could you make them all font-weight: 600;

image

**Comment by tyxla** [comment]
Thanks @rickybanister, that looks much better, here's an updated preview:

**Pull request merged**